NVHPC Fortran requires -Mbackslash flag

When I try to compile with NVHPC Fortran, it fails like this:
```
[ 58%] Built target mctc-lib-lib
[ 59%] Building Fortran object app/CMakeFiles/mctc-convert.dir/main.f90.o
NVFORTRAN-S-0026-Unmatched quote (/home/jehammond/MOLPRO/mctc-lib/app/main.f90: 284)
NVFORTRAN-S-0026-Unmatched quote (/home/jehammond/MOLPRO/mctc-lib/app/main.f90: 297)
NVFORTRAN/x86-64 Linux 24.7-0: compilation completed with severe errors
gmake[2]: *** [app/CMakeFiles/mctc-convert.dir/build.make:75: app/CMakeFiles/mctc-convert.dir/main.f90.o] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:279: app/CMakeFiles/mctc-convert.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

The solution is `FFLAGS=-Mbackslash FC=nvfortran cmake .. && cmake --build .`, i.e. to add the `-Mbackslash` flag.

Please add this to your CMake logic.  Thanks.

--- a/config/CMakeLists.txt
+++ b/config/CMakeLists.txt
@@ -37,6 +37,19 @@ install(
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
 )
 
+# Compiler-specific configurations 
+if(
+  CMAKE_Fortran_COMPILER_ID MATCHES "PGI"
+  OR CMAKE_Fortran_COMPILER_ID MATCHES "NVHPC"
+  OR CMAKE_Fortran_COMPILER_ID MATCHES "Flang"
+)
+  set(
+    CMAKE_Fortran_FLAGS
+    "${CMAKE_Fortran_FLAGS} -Mbackslash -Mallocatable=03"
+    PARENT_SCOPE
+  )
+endif()
+
 # Set build type as CMake does not provide defaults
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(

